cpp/examples/abm.cpp

```diff
@@ -33,128 +37,221 @@ epi::InfectionState determine_infection_state(double exposed, double infected, d
 
-    // 9% of population are childreen between 5 and 14 years.
-    int num_people_age5to14 = num_total_people * 0.09;
-    // The ncrement is 3 since in each loop 3 children are added.
-    for (auto i = 0; i < num_people_age5to14; i = i + 3) {
-        //Families with 2 children older than 5
-        create_household({epi::AbmAgeGroup::Age5to14, epi::AbmAgeGroup::Age5to14, epi::AbmAgeGroup::Age35to59,
-                          epi::AbmAgeGroup::Age35to59});
-        //Families with 1 child older than 5
-        create_household({epi::AbmAgeGroup::Age5to14, epi::AbmAgeGroup::Age15to34, epi::AbmAgeGroup::Age35to59});
-    }
 
-    //Add adults
-    // 23% of the population is between 15 and 34 years old.
-    int num_people_age15to24 = num_total_people * 0.23;
-    // Assumptions: Approx. 25% of people between 15 and 34 live with their children (they have been already added above)
-    // 8% lives with their parents, 36% lives alone and 31 % with their partner
-    for (auto i = 0; i < num_people_age15to24 * 0.08; ++i) {
-        //Person lives with parents
-        create_household({epi::AbmAgeGroup::Age15to34, epi::AbmAgeGroup::Age35to59, epi::AbmAgeGroup::Age35to59});
-    }
-    for (auto i = 0; i < num_people_age15to24 * 0.36; ++i) {
-        //Person lives alone
-        create_household({epi::AbmAgeGroup::Age15to34});
-    }
-    // Increment is 2 since in every loop two people between 15 and 34 years are added.
-    for (auto i = 0; i < num_people_age15to24 * 0.36; i = i + 2) {
-        //Person lives with partner
-        create_household({epi::AbmAgeGroup::Age15to34, epi::AbmAgeGroup::Age15to34});
-    }
-
-    // 35% of the population is between 35 and 59 years old.
-    int num_people_age35to59 = num_total_people * 0.35;
-    //Assumption: Approx. 33% of people between 35 and 59 lives with their younger children and approx. 11% lives with their teenager (have been already added above)
-    //39% lives with a partner and 17% lives alone
-    for (auto i = 0; i < num_people_age35to59 * 0.17; ++i) {
-        //Person lives alone
-        create_household({epi::AbmAgeGroup::Age35to59});
-    }
-    // Increment increases by 2 since in each loop 2 people between 35 and 39 are added.
-    for (auto i = 0; i < num_people_age35to59 * 0.39; i = i + 2) {
-        //Person lives with partner
-        create_household({epi::AbmAgeGroup::Age35to59, epi::AbmAgeGroup::Age35to59});
+/**
+ * Function which constructs a household group which has a single member to represent them all, e.g. all people have the same age distribution. 
+ * @param age_dist A vector with the amount of people in each age group
+ * @param number_of_people The total amount of people living in this household group.
+ * @param number_of_hh The number of households in this household group.
+ * @return householdGroup A Class Household Group.
+ */
+epi::HouseholdGroup make_household_group_from_one_member(const epi::HouseholdMember& member, int number_of_people, int number_of_hh){
```

Comment:
this sounds like the households only have one member. The should be a better name. Maybe `make_uniform_households` and `make_family_households` ?